src/Fabulous.Core/ViewHelpers.fs

```diff
@@ -92,6 +92,37 @@ module SimplerHelpers =
                 false // Do not let the timer trigger a second time
             ))
 
+    /// Looks for a view element with the given Automation ID in the view hierarchy.
+    /// This function is not optimized for efficiency and may execute slowly.
+    let rec tryFindViewElement automationId (element:ViewElement) =
+        let elementAutomationId = element.TryGetAttribute<string>("AutomationId")
+        match elementAutomationId with
+        | ValueSome automationIdValue when automationIdValue = automationId -> Some element
+        | _ ->
+            let childElements =
+                match element.TryGetAttribute<ViewElement>("Content") with
+                | ValueSome content -> [| content |]
+                | ValueNone ->
+                    match element.TryGetAttribute<ViewElement[]>("Pages") with
+                    | ValueSome pages -> pages
+                    | ValueNone ->
+                        match element.TryGetAttribute<ViewElement[]>("Children") with
+                        | ValueNone -> [||]
+                        | ValueSome children -> children
+
+            childElements
+            |> Seq.map (fun e -> e |> tryFindViewElement automationId)
+            |> Seq.filter (fun e -> e.IsSome)
+            |> Seq.map (fun e -> e.Value)
```

Comment:
Is this mapping useful? You are only interested in the first element after all. So you could take the first and return the value.

Reply:
I'm doing this 2nd map here because the type returned by Seq.filter is `Seq<ViewElement option>`, so Seq.tryHead will return a `ViewElement option option`.

Without this map, I would need to match on the `ViewElement option option` and in the case `tryHead` found something, convert the `ViewElement option` to `ViewElement`.
I think it's faster to read and write the way I did.

Also the advantage of Seq here is that it won't evaluate the whole collection (like IEnumerable and LINQ in C#), it'll try each item one by one until it finds a matching one.

Reply:
If it works like this then I didn't say anything. Was just curious about it because if it would work like LINQ it would be bad for performance.

Reply:
You could use Seq.choose here instead of Seq.map and Seq.filter, which would auto-unwrap the option for you and reduce the option nesting.